Calibration/HcalCalibAlgos/macros/CalibTree.C

```diff
@@ -0,0 +1,1103 @@
+			   << std::endl;
+      if (loop==0) {
+	h_pbyE->Fill(ratio, evWt);
+        h_Ebyp_bfr->Fill(t_ieta, ratio, evWt);
+      }
+      if (last){
+        h_Ebyp_aftr->Fill(t_ieta, ratio, evWt);
+      }
+      bool l1c(true);
+      if (applyL1Cut != 0) l1c = ((t_mindR1 >= l1Cut) || 
+				  ((applyL1Cut == 1) && (t_DataType == 1)));
+      if ((rmin >=0 && ratio > rmin) && (rmax >= 0 && ratio < rmax) && l1c) {
+	for (unsigned int idet=0; idet<(*t_DetIds).size(); idet++) {
+	  if (selectPhi((*t_DetIds)[idet])) {
+	    unsigned int id    = (*t_DetIds)[idet];
+	    unsigned int detid = truncateId(id);
+	    double hitEn=0.0;
+	    if (debug) std::cout << "idet " << idet << " detid/hitenergy : " 
+				 << std::hex << (*t_DetIds)[idet] << ":" 
+				 << detid << "/" << (*t_HitEnergies)[idet] 
+				 << std::endl;
+	    if (Cprev.find(detid) != Cprev.end()) 
+	      hitEn = Cprev[detid].first * (*t_HitEnergies)[idet];
+	    else 
+	      hitEn = (*t_HitEnergies)[idet];
+	    if (cFactor_ != 0) hitEn *= cFactor_->getCorr(t_Run,id);
+	    double Wi  = evWt * hitEn/Etot;
+	    double Fac = (inverse) ? (pufac*Etot/(pmom-t_eMipDR)) : 
+	      ((pmom-t_eMipDR)/(pufac*Etot));
+	    double Fac2= Wi*Fac*Fac;
+	    TH1D* hist(0);
+	    std::map<unsigned int,TH1D*>::iterator itr = histos.find(detid);
+	    if (itr != histos.end()) hist = itr->second;
+	    if (debug) std::cout << "Det Id " << std::hex << detid << std::dec 
+				 << " " << hist << std::endl;
+	    if (hist != 0) hist->Fill(Fac, Wi);//////histola
+	    Fac       *= Wi;
+	    if (SumW.find(detid) != SumW.end() ) {
+	      Wi  += SumW[detid].fact0;
+	      Fac += SumW[detid].fact1;
+	      Fac2+= SumW[detid].fact2;
+	      int kount = SumW[detid].kount + 1;
+	      SumW[detid]   = myEntry(kount,Wi,Fac,Fac2); 
+	      nTrks[detid] += evWt;
+	    } else {
+	      SumW.insert(std::pair<unsigned int,myEntry>(detid,myEntry(1,Wi,Fac,Fac2)));
+	      nTrks.insert(std::pair<unsigned int,unsigned int>(detid, evWt));
+	    }
+	  }
+	}
+      }
+    }
+  }
+  if (debug) std::cout << "# of Good Tracks " << ntkgood << " out of "
+		       << nentries << std::endl;
+  if (loop==0) {
+    h_pbyE->Write("h_pbyE");
+    h_Ebyp_bfr->Write("h_Ebyp_bfr");
+  }
+  if (last) {
+    h_Ebyp_aftr->Write("h_Ebyp_aftr");
+  }
+
+  std::map<unsigned int, std::pair<double,double> > cfactors;
+  unsigned int kount(0), kountus(0);
+  double       sumfactor(0);
+  for (std::map<unsigned int,TH1D*>::iterator itr = histos.begin();
+       itr != histos.end(); ++itr) {
+    if (writeHisto) {
+      std::pair<double,double> result_write = fitMean(itr->second, 0);
+      (itr->second)->Write();
+    }
+    int subdet = ((itr->first) >> 25) & (0x7);
+    int ieta   = ((itr->first) >> 10) & (0x1FF);
+    if (debug) std::cout << "DETID :" << subdet << "  IETA :" << ieta
+			 << " HIST ENTRIES :" << (itr->second)->GetEntries()
+			 << std::endl;
+  }
+
+  for (std::map<unsigned int,TH1D*>::iterator itr = histos.begin();

```

Comment:
this could be a `const_iterator`, right?
or use `auto` together with `cbegin()`
applies also to iterators below